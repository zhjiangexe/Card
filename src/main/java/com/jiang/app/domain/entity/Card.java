package com.jiang.app.domain.entity;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import java.io.Serializable;
import java.math.BigDecimal;

@Data
@Entity
@Table(name = "CARD")
public class Card implements Serializable {
  @Id
  private String cardNo;

  private String userName;

  private BigDecimal balance;

  @Column(name = "LIMIT_VALUE")
  private BigDecimal limit;

  public Card() {
  }

  public Card(String userName, String cardNo, BigDecimal balance, BigDecimal limit) {
    this.cardNo = cardNo;
    this.userName = userName;
    this.balance = balance;
    this.limit = limit;
  }

}
