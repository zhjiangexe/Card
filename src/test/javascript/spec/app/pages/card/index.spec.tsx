import React from 'react'
import { shallow, mount } from 'enzyme'

import Card from 'app/pages/card'
import Add from 'app/pages/card/add'
import List from 'app/pages/card/list'

describe('Card', () => {
  let mountedWrapper
  const devProps = {}
  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = mount(
        <Card/>
      )
    }
    return mountedWrapper
  }

  beforeEach(() => {
    mountedWrapper = undefined
  })

  it('Renders a Header component with List, Add components.', () => {
    const component = wrapper()

    expect(component).toMatchSnapshot()
    expect(component.find(List).length).toEqual(1)
    expect(component.find(Add).length).toEqual(1)
  })

})
