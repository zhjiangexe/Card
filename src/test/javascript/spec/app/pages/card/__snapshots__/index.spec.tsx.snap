// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Renders a Header component with List, Add components. 1`] = `
<cardPage>
  <Container
    tag="div"
  >
    <div
      className="container"
    >
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col"
            >
              <h1>
                Credit Card System
              </h1>
            </div>
          </Col>
        </div>
      </Row>
      <Row
        tag="div"
      >
        <div
          className="row"
        >
          <Col
            md="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-md-6"
            >
              <add
                addCard={[Function]}
              >
                <div>
                  <h2>
                    Add
                  </h2>
                  <Form
                    tag="form"
                  >
                    <form
                      className=""
                    >
                      <FormGroup
                        tag="div"
                      >
                        <div
                          className="form-group"
                        >
                          <Label
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            <label
                              className=""
                            >
                              Name
                            </label>
                          </Label>
                          <Input
                            name="userName"
                            onChange={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              className="form-control"
                              name="userName"
                              onChange={[Function]}
                              type="text"
                              value=""
                            />
                          </Input>
                        </div>
                      </FormGroup>
                      <FormGroup
                        tag="div"
                      >
                        <div
                          className="form-group"
                        >
                          <Label
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            <label
                              className=""
                            >
                              Card number
                            </label>
                          </Label>
                          <Input
                            name="cardNo"
                            onChange={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              className="form-control"
                              name="cardNo"
                              onChange={[Function]}
                              type="text"
                              value=""
                            />
                          </Input>
                        </div>
                      </FormGroup>
                      <FormGroup
                        tag="div"
                      >
                        <div
                          className="form-group"
                        >
                          <Label
                            tag="label"
                            widths={
                              Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                            }
                          >
                            <label
                              className=""
                            >
                              Limit
                            </label>
                          </Label>
                          <Input
                            name="limit"
                            onChange={[Function]}
                            type="text"
                            value=""
                          >
                            <input
                              className="form-control"
                              name="limit"
                              onChange={[Function]}
                              type="text"
                              value=""
                            />
                          </Input>
                        </div>
                      </FormGroup>
                      <FormGroup
                        tag="div"
                      >
                        <div
                          className="form-group"
                        >
                          <Button
                            color="primary"
                            onClick={[Function]}
                            tag="button"
                          >
                            <button
                              aria-label={null}
                              className="btn btn-primary"
                              onClick={[Function]}
                              type="button"
                            >
                              Add
                            </button>
                          </Button>
                        </div>
                      </FormGroup>
                    </form>
                  </Form>
                </div>
              </add>
            </div>
          </Col>
          <Col
            md="6"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <div
              className="col-md-6"
            >
              <listBlock
                cards={Array []}
              >
                <div>
                  <h2>
                    Existing Cards
                  </h2>
                  <Table
                    responsive={true}
                    responsiveTag="div"
                    striped={true}
                    tag="table"
                  >
                    <div
                      className="table-responsive"
                    >
                      <table
                        className="table table-striped"
                      >
                        <thead>
                          <tr>
                            <th>
                              Name
                            </th>
                            <th>
                              Card Number
                            </th>
                            <th>
                              Balance
                            </th>
                            <th>
                              Limit
                            </th>
                          </tr>
                        </thead>
                        <tbody>
                          <tr>
                            <td
                              colSpan={4}
                            >
                              There are no cards.
                            </td>
                          </tr>
                        </tbody>
                      </table>
                    </div>
                  </Table>
                </div>
              </listBlock>
            </div>
          </Col>
        </div>
      </Row>
    </div>
  </Container>
</cardPage>
`;
